src/game_event.cpp

```diff
@@ -557,7 +557,12 @@ void Game_Event::MoveTypeRandom() {
 	int last_direction = GetDirection();
 	switch (rand() % 6) {
 	case 0:
-		stop_count -= rand() % stop_count;
+		if (stop_count > 0) {
+			stop_count -= rand() % stop_count;
+			if (stop_count < 0) {
+				stop_count = 0;
+			}
+		}
 		return;
```

Comment:
Just leaving this here in case you did miss it:
`2016-04-24 21:49:05 <zegeri> Ghabry: It should be "stop_count -= rand() % (stop_count + 1);"`